src/models/history/tree-manager.ts

```diff
@@ -46,22 +49,37 @@ interface IFirestoreHistoryInfo {
 
 interface IMainDocument extends TreeAPI {
   key: string;
+  uid: string;
   metadata: IDocumentMetadata;
 }
 
+export enum HistoryStatus {
+  HISTORY_ERROR,
+  FINDING_HISTORY_LENGTH,
+  NO_HISTORY,
+  HISTORY_LOADED,
+  HISTORY_LOADING,
+}
+
 export const TreeManager = types
 .model("TreeManager", {
   document: CDocument,
   undoStore: UndoStore,
-  // This is not volatile, so the TreeManager actions can modify it directly
+  // This is not volatile, so the TreeManager actions can modify it directly.
+  // If it was volatile, it would be its own MST tree, and the actions in this
+  // TreeManager model can not modify other MST trees.
   // When the history is serialized, we'll only serialize the CDocument, so it
   // shouldn't matter that these uncompleted history entries are part of the
-  // tree.
+  // state of the TreeManager.
   activeHistoryEntries: types.array(HistoryEntry)
 })
 .volatile(self => ({
   trees: {} as Record<string, TreeAPI>,
-  currentHistoryIndex: 0,
+  // This name is deceiving. This property is the number of history entries
+  // that have been applied to the documents. So 0 means no history entries
+  // are applied.
+  currentHistoryIndex: 0 as number | undefined,
```

Comment:
Would it make more sense to call this `historyEventsApplied` or something similar?

Reply:
Yes that is a good name. I didn't want to rename it in this PR because I thought it would impact more files. But looking again, it only impacts the files of this PR. So I'll go ahead and change it.